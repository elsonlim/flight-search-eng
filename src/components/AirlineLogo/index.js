import React, { Component } from 'react';
import { Image } from 'semantic-ui-react'; 

class AirlineLogo extends Component {

    // { origin: 'USA', IATA: 'DL' },
    // { origin: 'AUS', IATA: 'QAN' },
    // { origin: 'SGP', IATA: 'SQ' },
    // { origin: 'HKG', IATA: 'CX' },
    // { origin: 'JPN', IATA: 'JAL' },
    // { origin: 'JPN', IATA: 'ANA' },
    // { origin: 'TWN', IATA: 'BR' },
    // { origin: 'VNM', IATA: 'EVA' },

    getImage(code) {
        const airlineLogo = {
            DL: 'dl.svg',
            QAN: 'qan.svg',
            SQ: 'sq.svg',
            CX: 'cx.svg',
            JAL: 'jal.svg',
            ANA: 'ana.svg',
            EVA: 'eva.svg',
        }
        
        return `/airlineLogo/${airlineLogo[code]}`;
    }

    render() {
        return (
            <div>
                <Image rounded={true} size={'tiny'} src={this.getImage(this.props.airlineLogo)} />
            </div>
        );
    }
}

export default AirlineLogo;