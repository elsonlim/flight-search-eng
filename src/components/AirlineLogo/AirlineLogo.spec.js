import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import AirlineLogo from './index';

describe('AirlineLogo', () => {

    let props;
    let wrapper;
    
    beforeEach(() => {
        props = {
            airlineLogo: 'DL'
        }
    });

    it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<AirlineLogo {...props} />, div);
		ReactDOM.unmountComponentAtNode(div);
    });
    
    describe('snapshot test', () => {
		test('show airline logo', () => {
			const tree = renderer.create(<AirlineLogo {...props} />).toJSON();
			expect(tree).toMatchSnapshot();
		});
	});
});
