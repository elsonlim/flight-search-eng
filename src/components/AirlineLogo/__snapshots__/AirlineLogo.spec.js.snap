// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AirlineLogo snapshot test show airline logo 1`] = `
<div>
  <img
    className="ui tiny rounded image"
    src="/airlineLogo/dl.svg"
  />
</div>
`;
